packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,401 @@
+/// and when to regenerated the child image.
+///
+/// When the value of [rasterize] is true, The [RasterWidget] will convert the child scene into
+/// an image. This image will be drawn in place of the children until [rasterize] is false.
+/// Any changes to the widget tree that would normally cause it to repaint are ignored.
+///
+/// To force [RasterWidget] to recreate the child image, call [clear].
+class RasterWidgetController extends ChangeNotifier {
+  /// Create a new [RasterWidgetController].
+  ///
+  /// By default, [rasterize] is `false` and cannot be `null`.
+  RasterWidgetController({
+    bool rasterize = false
+  }) : _rasterize = rasterize;
+
+  /// Reset the raster held by any listening [RasterWidget].
+  ///
+  /// This has no effect if [rasterize] is `false`.
+  void clear() {
+    notifyListeners();
+  }
+
+  /// Whether a rasterized version of this render objects child is drawn in
+  /// place of the child.
+  bool get rasterize => _rasterize;
+  bool _rasterize;
+  set rasterize(bool value) {
+    if (value == rasterize) {
+      return;
+    }
+    _rasterize = value;
+    notifyListeners();
+  }
+}
+
+/// A delegate which the [RasterWidget] can use to fallback to regular rendering
+/// if a platform view is present in the layer tree.
+///
+/// Consumers of [RasterWidget] should almost never use this delegate. For the most part,
+/// the raster widget only functions as a performance improvement. If a platform view is
+/// present, the performance improving qualities aren't possible and using this API is
+/// pointless.
+///
+/// Instead, this interface is useful if a generic/reusable widget is being created which
+/// may include a platform view and it needs to handle this transparently. For example, the
+/// framework uses this for the zoom page transition so that navigating to a page shows the same
+/// animation whether or not there is a platform view.
+abstract class RasterWidgetFallbackDelegate {
+  /// const constructor so that subclasses can be const.
+  const RasterWidgetFallbackDelegate();
+
+  /// Paint the child via [painter], applying any effects that would have been painted
+  /// with the [RasterWidgetDelegate].
+  ///
+  /// The [offset] and [size] are the location and dimensions of the render object.
+  void paintFallback(PaintingContext context, Offset offset, Size size, PaintingContextCallback painter);
+}
+
+/// A widget that replaces its child with a rasterized version of the child.
+///
+/// By default, the child is drawn as is. The default [delegate] simply scales
+/// down the image by the current device pixel ratio and paints it into the
+/// canvas. How this image is drawn can be customized by providing a new
+/// subclass of [RasterWidgetDelegate] to the [delegate] argument.
+///
+/// Caveats:
+///
+/// The contents of platform views cannot be captured by a raster
+/// widget. If a platform view is encountered, then nothing will be rendered unless
```

Comment:
This is at odds with the documentation at ignorePlatformViews, which says an error is thrown. I think throwing an error is preferred over silently drawing nothing.